independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -1126,6 +1149,7 @@ private void addSyntheticBean(BeanInfo bean) {
             }
         }
         beans.add(bean);
+        beansByType = null;
```

Comment:
This should not be necessary - synthetic beans can only be added before initialization, i.e. before the `BeanDeployment.init()` method is called.